src/components/bar-chart/bar-chart.js

```diff
@@ -1,25 +1,55 @@
 import template from "./bar-chart.html";
 import Chart from "chart.js";
 
+import { CurrencyConverterUAH } from "../../services/currency-converter-uah";
+
 export class BarChartComponent {
   constructor(mountPoint, props) {
     this.mountPoint = mountPoint;
     this.props = props;
   }
 
   querySelectors() {
-    this.barChartCtx = this.mountPoint.querySelector(".chart__visual");
+    this.chartCtx = this.mountPoint.querySelector(".chart__visual");
+  }
+
+  update(data) {
+    this.dataset = this.addCurrData(this.dataset, data);
+    this.drawChanged();
   }
 
-  createBarChart() {
-    this.barChart = new Chart(this.barChartCtx, {
+  createFromList(list) {
+    this.dataset = list.reduce(this.addCurrData, this.dataset);
+    this.drawChanged();
+  }
+
+  addCurrData(accum, item) {
+    let data = accum;
```

Comment:
you're mutating `accum` at the bottom

Reply:
you can use either `const data = Object.assign({}, accum)` or es2018 object rest-spread `const data = { ...accum }`